src/ethereum/forks/amsterdam/vm/interpreter.py

```diff
@@ -263,10 +209,19 @@ def process_create_message(message: Message) -> Evm:
     # added to SELFDESTRUCT by EIP-6780.
     mark_account_created(state, message.current_target)
 
-    parent_frame = get_parent_frame(message)
-    create_frame = create_child_frame(parent_frame)
+    block_access_index = get_block_access_index(
+        message.block_env.block_state_changes
+    )
+
+    increment_nonce(state, message.current_target)
+    nonce_after = get_account(state, message.current_target).nonce
+    track_nonce_change(
+        message.state_changes,
```

Comment:
If we make the `_block_access_index` directly available in every child frame, we would not need to pass it as a function parameter